Nrf54l15 features doc (#14956)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf54l/features.rst

.. _ug_nrf54l_features:

Features of the nRF54L15 PDK
############################

.. contents::
   :local:
   :depth: 2

The nRF54L15 PDK embeds an Arm Cortex-M33 processor with multiprotocol 2.4 GHz transceiver and supports Bluetooth® 5.4.

For additional information, see the following documentation:

> 

Author: Linked a zephyr page for now as to my knowledge no other docs are publicly available

* :ref:`ug_nrf54l15_gs`.
* :ref:`installation` and :ref:`configuration_and_build` documentation to install the |NCS| and learn more about its development.environment.

Device Firmware Update and bootloader support
*********************************************

The nRF54L15 PDK supports MCUboot.
Currently the support is experimental limited to SW cryptography.
Additionally, a single image pair is supported (dual-bank DFU for cpuapp).

.. note::
   MCUboot as the second stage bootloader and serial recovery is not supported yet.

The supported DFU solution is mcumgr (Bluetooth LE, serial), and you can test it with the :ref:`zephyr:smp_svr_sample`, for which Bluetooth LE and serial channels are available.
You can build it with support for child images and the partition manager for the ``nrf54l15pdk_nrf54l51_cpuapp`` build target.

The support includes secondary image slot in the external SPI NOR flash, which can be tested by running the following command:

   .. code-block:: console

>       west build  -b nrf54l15pdk_nrf54l15_cpuapp -d build/smp_svr_54l_3 zephyr/samples/subsys/mgmt/mcumgr/smp_svr -T sample.mcumgr.smp_svr.bt.nrf54l15pdk.ext_flash

Author: Yeah, the plan is to move it to "developing with" when the page is created. Current solution is temporary

> .. _ug_nrf54l_features:

Reviewer: Sorry, missed this - to be consistent with other features pages
```suggestion
.. features_nrf54l:
```

For more information about the multiprotocol feature, see :ref:`ug_multiprotocol_support`.

See the :ref:`ug_matter` user guide for information about how to work with Matter applications.
To start developing, check the :ref:`matter_samples`.

Thread
======

.. include:: ../../../protocols/thread/index.rst
   :start-after: thread_intro_start
   :end-before: thread_intro_end

You can read more about other available platform designs for Thread on the :ref:`OpenThread architectures page<ug_thread_architectures>`.
For more information about the multiprotocol feature, see :ref:`ug_multiprotocol_support`.

To start developing, check out the :ref:`openthread_samples`.

Near Field Communication
========================

Near Field Communication (NFC) is a technology for wireless transfer of small amounts of data between two devices that are in close proximity.
The range of NFC is typically less than 10 cm.

Refer to the :ref:`ug_nfc` page for general information.
See the :ref:`nfc_samples` and :ref:`lib_nfc` for the samples and libraries that the |NCS| provides.

> MCUboot bootloader support

Author: @nvlsianpu could you please review the reworked bootloader and DFU part? thanks!

**************************

The nRF54L15 PDK supports MCUboot as its bootloader, in the experimental phase.
This means the following:

  * Only software cryptography is supported.
  * Single image pair is supported for dual-bank Device Firmware Update (DFU) targeted at the CPU application (the ``nrf54l15pdk_nrf54l51_cpuapp`` build target).
  * MCUboot can be configured as a first-stage bootloader (second-stage bootloader functionality is not yet available).
  * Serial recovery mode is also not yet supported.

Supported DFU protocols
=======================

> The DFU process in the nRF54L15 PDK uses the MCUmgr protocol, exclusively.

Reviewer 2: I'm not sure whether we really need to emphasize that only MCUmgr works. Other DFU layer might works as well, we just don't check them.

Author: Got it - will remove 'exclusively' part, thanks!

.. _ug_nrf54l_features:

Features of the nRF54L15 PDK
############################

.. contents::
   :local:
   :depth: 2

The nRF54L15 PDK embeds an Arm Cortex-M33 processor with multiprotocol 2.4 GHz transceiver and supports Bluetooth® 5.4.

For additional information, see the following documentation:

* :ref:`ug_nrf54l15_gs`.
* :ref:`installation` and :ref:`configuration_and_build` documentation to install the |NCS| and learn more about its development.environment.

Supported protocols
*******************

The nRF54L15 PDK supports Bluetooth Low Energy (LE), proprietary protocols (including Enhanced ShockBurst), Matter, and Thread.

Bluetooth Low Energy
====================

The Bluetooth Low Energy (LE) radio is designed for very low power operation.
When you develop a Bluetooth LE application, you must use the Bluetooth software stack.
This stack is split into two core components: the Bluetooth Host and the Bluetooth LE Controller.

> The :ref:`ug_ble_controller` user guide contains more information about the two available Bluetooth LE Controllers and instructions for switching between them.

Reviewer 3: ```suggestion
The :ref:`ug_ble_controller` user guide contains more information about the two available Bluetooth LE Controllers, and instructions for switching between them.
```

> See :ref:`zephyr:bluetooth` for documentation on the Bluetooth Host and open source Bluetooth LE Controller.

Reviewer 3: ```suggestion
See the :ref:`zephyr:bluetooth` section of the Zephyr documentation for information on the Bluetooth Host and open source Bluetooth LE Controller.
```

The |NCS| contains :ref:`ble_samples` that can be run on the nRF54L15 PDK device.
In addition, you can run the :ref:`zephyr:bluetooth-samples` that are included from Zephyr.

For available libraries, see :ref:`lib_bluetooth_services` (|NCS|) and :ref:`zephyr:bluetooth_api` (Zephyr).

Enhanced ShockBurst
===================

.. include:: ../../../protocols/esb/index.rst
   :start-after: esb_intro_start
   :end-before: esb_intro_end

See the :ref:`ug_esb` user guide for information about how to work with Enhanced ShockBurst.
To start developing, check out the :ref:`esb_ptx` and :ref:`esb_prx` samples.

Matter
======

.. include:: ../../../protocols/matter/index.rst
   :start-after: matter_intro_start
   :end-before: matter_intro_end

Matter in the |NCS| supports the *System-on-Chip, multiprotocol* platform design for the nRF54L15 SoC using Matter over Thread.
You can read more about other available platform designs for Matter on the :ref:`Matter platform design page<ug_matter_overview_architecture_integration_designs>`.

See the :ref:`ug_matter` user guide for information about how to work with Matter applications.

> To start developing, check out :ref:`matter_samples`.

Reviewer 3: ```suggestion
To start developing, check out the :ref:`matter_samples`.
```

> For more information about the multiprotocol feature, see :ref:`ug_multiprotocol_support`.

Reviewer 3: This could probably be moved to be part of the paragraph with Lines 54-55.

The following figure demonstrates the architecture:

.. _nrf54l15_platform_multi_figure:

.. figure:: images/thread_platform_design_nrf54l15_multi.svg
   :alt: Multiprotocol Thread and Bluetooth LE architecture (nRF54L15)

   Multiprotocol Thread and Bluetooth LE architecture on the nRF54L15 SoC

Matter over Thread
------------------

In this design:

* The Matter stack, the OpenThread stack, and the Bluetooth LE stack run on single core of a single nRF54L15 SoC.
* The application core runs the Matter stack, the OpenThread stack, the component of the OpenThread stack that is related to the 802.15.4 IEEE Radio Driver and a part of the Bluetooth LE Controller.

  .. note::
      The PSA crypto API level 3 for storing security components is not yet implemented on the Secure Domain.

Refer to the :ref:`nrf54l15_platform_multi_figure` figure to see the architecture of the SoC.
The Global Domain is not included.

Thread
======

.. include:: ../../../protocols/thread/index.rst
   :start-after: thread_intro_start
   :end-before: thread_intro_end

Thread in the |NCS| supports the *System-on-Chip, single protocol* and *System-on-Chip, multiprotocol* platform designs for the nRF54L15 SoC.

> You can read more about other available platform designs for Thread on the :ref:`OpenThread architectures page<ug_thread_architectures>`.
> To start developing, refer to the :ref:`openthread_samples`.

Reviewer 3: ```suggestion
You can read more about other available platform designs for Thread on the :ref:`OpenThread architectures page<ug_thread_architectures>`.
For more information about the multiprotocol feature, see :ref:`ug_multiprotocol_support`.

To start developing, check out the :ref:`openthread_samples`.
```

Near Field Communication
========================

Near Field Communication (NFC) is a technology for wireless transfer of small amounts of data between two devices that are in close proximity.

> The range of NFC is typically <10 cm.

Reviewer 3: ```suggestion
The range of NFC is typically less than 10 cm.
```

Refer to the :ref:`ug_nfc` page for general information.

> See the :ref:`nfc_samples` and :ref:`lib_nfc` for lists of samples and libraries that the |NCS| provides.

Reviewer 3: ```suggestion
See the :ref:`nfc_samples` and :ref:`lib_nfc` for the samples and libraries that the |NCS| provides.
```
